src/main/java/gregtech/common/pipelike/fluidpipe/BlockFluidPipe.java

```diff
@@ -110,35 +110,26 @@ public boolean isHoldingPipe(EntityPlayer player) {
     }
 
     @Override
-    public void onEntityCollision(World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state, @Nonnull Entity entityIn) {
+    public void onEntityCollision(@Nonnull World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state, @Nonnull Entity entityIn) {
         if (worldIn.isRemote) return;
-        if (entityIn instanceof EntityLivingBase && entityIn.world.getTotalWorldTime() % 20 == 0L) {
-            EntityLivingBase entityLiving = (EntityLivingBase) entityIn;
-            TileEntityFluidPipe pipe = (TileEntityFluidPipe) getPipeTileEntity(worldIn, pos);
-            if(!(pipe instanceof TileEntityFluidPipeTickable))
-                return;
-            List<Integer> temps = new ArrayList<>();
-            for (FluidTank tank : ((TileEntityFluidPipeTickable) pipe).getFluidTanks()) {
-                if (tank.getFluid() != null && tank.getFluid().amount > 0) {
-                    temps.add(tank.getFluid().getFluid().getTemperature(tank.getFluid()));
+        TileEntityFluidPipe pipe = (TileEntityFluidPipe) getPipeTileEntity(worldIn, pos);
+        if (pipe instanceof TileEntityFluidPipeTickable && ((TileEntityFluidPipeTickable) pipe).getOffsetTimer() % 10 == 0) {
+            if (entityIn instanceof EntityLivingBase) {
+                // apply temperature damage for hottest and coldest pipe
+                int maxTemperature = Integer.MIN_VALUE;
+                int minTemperature = Integer.MAX_VALUE;
+                for (FluidTank tank : ((TileEntityFluidPipeTickable) pipe).getFluidTanks()) {
+                    if (tank.getFluid() != null && tank.getFluid().amount > 0) {
+                        maxTemperature = Math.max(maxTemperature, tank.getFluid().getFluid().getTemperature(tank.getFluid()));
+                        minTemperature = Math.min(minTemperature, tank.getFluid().getFluid().getTemperature(tank.getFluid()));
+                    }
+                }
+                if (maxTemperature != Integer.MIN_VALUE) {
+                    EntityDamageUtil.applyTemperatureDamage((EntityLivingBase) entityIn, maxTemperature, 1.0F, 5);
+                }
+                if (minTemperature != Integer.MAX_VALUE) {
+                    EntityDamageUtil.applyTemperatureDamage((EntityLivingBase) entityIn, minTemperature, 1.0F, 5);
```

Comment:
This will be an issue if there is only 1 fluid in the pipe. The max temperature and the min temperature will be set to the same thing, and so `applyTemperatureDamage` will be called twice, because `maxTemperature` and `minTemperature` are both not equal to the Max/Min values.